includes/Checker/Abstract_Check_Runner.php

```diff
@@ -255,45 +264,18 @@ private function get_shared_preparations( array $checks ) {
 	 *
 	 * @return array An array map of check slugs to Check instances.
 	 *
-	 * @throws Exception Thrown exception when a runtime check is requested and the plugin inactive.
+	 * @throws Exception Thrown when invalid flag is passed, or Check slug does not exist.
 	 */
 	final public function get_checks_to_run() {
-		$check_slugs   = $this->get_check_slugs();
-		$all_checks    = $this->get_checks_instance()->get_checks();
-		$plugin_active = is_plugin_active( $this->get_plugin_basename() );
-
-		if ( ! empty( $check_slugs ) ) {
-			// Get the check instances based on the requested checks.
-			$checks_to_run = array_intersect_key( $all_checks, array_flip( $check_slugs ) );
-
-			// Check the following conditions if at least one runtime check is requested.
-			if ( $this->has_runtime_check( $checks_to_run ) ) {
-				// Throw an error if the plugin is not active.
-				if ( ! $plugin_active ) {
-					throw new Exception( __( 'Runtime checks cannot be run against inactive plugins.', 'plugin-check' ) );
-				}
+		$check_slugs = $this->get_check_slugs();
+		$check_flags = Check_Repository::TYPE_STATIC;
 
-				// Throw and error if the runner was not initialized early and the runtime environment was not set up.
-				if ( ! $this->initialized_early ) {
-					throw new Exception( __( 'Runtime checks cannot be run as the runtime environment was not set up.', 'plugin-check' ) );
-				}
-			}
-		} else {
-			// Run all checks for the plugin.
-			$checks_to_run = $all_checks;
-
-			// Only run static checks if the plugin is inactive or the runtime environment was not set up.
-			if ( ! $plugin_active || ! $this->initialized_early ) {
-				$checks_to_run = array_filter(
-					$checks_to_run,
-					function ( $check ) {
-						return ! $check instanceof Runtime_Check;
-					}
-				);
-			}
+		// Check if conditions are met in order to perform Runtime Checks.
+		if ( $this->initialized_early ) {
```

Comment:
I think you missed the check for whether the plugin is active here, which is now missing.